src/Sylius/Bundle/CoreBundle/Tests/Functional/StateMachine/OrderWorkflowTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Functional\StateMachine;
+
+use Sylius\Bundle\CoreBundle\StateMachine\StateMachineInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\Customer;
+use Sylius\Component\Core\Model\Order;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+final class OrderWorkflowTest extends KernelTestCase
+{
+    private OrderInterface $order;
+
+    public function setup(): void
+    {
+        parent::setUp();
+        $sequenceRepository = $this->createMock(RepositoryInterface::class);
+        $sequenceRepository
+            ->method('findOneBy')
+            ->willReturn(null)
+        ;
+
+        self::getContainer()->set('sylius.repository.order_sequence', $sequenceRepository);
+
+        $channel = $this->createMock(ChannelInterface::class);
+        $customer = $this->createMock(Customer::class);
+        $order = new Order();
+        $order->setChannel($channel);
+        $order->setCustomer($customer);
+
+        $this->order = $order;
+    }
+
+    /** @test */
+    public function it_applies_new_state_for_order_cart_state(): void
+    {
+        $stateMachine = $this->getStateMachine();
+        $order = $this->order;
+
+        $this->assertSame('cart', $order->getState());
+
+        $stateMachine->apply($order, 'sylius_order', 'create');
```

Comment:
Could be easily changed to consts, same goes for other states/transitions/graph names